causal_testing/data_collection/data_collector.py

```diff
@@ -73,7 +73,10 @@ def filter_valid_data(self, data: pd.DataFrame, check_pos: bool = True) -> pd.Da
         size_diff = len(data) - len(satisfying_data)
         if size_diff > 0:
             logger.warning(
-                f"Discarded {size_diff}/{len(data)} values due to constraint violations.\n" f"For example{unsat_core}"
+                "Discarded %s/%s values due to constraint violations.\n" "For example%s",
```

Comment:
Looking at these [w1203](https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html) things, I prefer them as F-strings. I think the readability of F-strings outweighs the performance / other potential benefits of %. More discussion here https://github.com/PyCQA/pylint/issues/2395

Reply:
This would mean disabling some checks for a clean linter run.

Reply:
I agree as well, F-Strings are much clearer and I'm not sure there would be a noticeable performance impact. So I'll happily revert them to F-Strings for now!

Interesting read as well cheers Bob :)